lib/config/configuration.go

```diff
@@ -1898,7 +1902,18 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 		if len(resources) < 1 {
 			return trace.BadParameter("no resources found: %q", clf.BootstrapFile)
 		}
-		cfg.Auth.Resources = resources
+		cfg.Auth.BootstrapResources = resources
+	}
+
+	if clf.ApplyOnStartupFile != "" {
+		resources, err := ReadResources(clf.ApplyOnStartupFile)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if len(resources) < 1 {
+			return trace.BadParameter("no resources found: %q", clf.ApplyOnStartupFile)
+		}
```

Comment:
Looking above, bootstrap also treats empty file as error but I'm not 100% sure. Maybe to help users prevent unexpected situations where they expect this file to not be empty? I'd think that if you supply a file with empty resources we just don't apply anything. Anyway, I'm ok to keep it consistent with how --bootstrap behaves, just something that crossed my mind.

Reply:
Yeah, I really prefer this to fail early in case of wrong path/empty file instead of starting without the intended resources. For example, in Kubernetes a missing volume mount would block the rollout.